libcontainer/console.go

```diff
@@ -11,5 +16,59 @@ type Console interface {
 	Path() string
 
 	// Fd returns the fd for the master of the pty.
-	Fd() uintptr
+	File() *os.File
+}
+
+const (
+	TerminalInfoVersion uint32 = 201610041
+	TerminalInfoType    uint8  = 'T'
+)
+
+// TerminalInfo is the structure which is passed as the non-ancillary data
+// in the sendmsg(2) call when runc is run with --console-socket. It
+// contains some information about the container which the console master fd
+// relates to (to allow for consumers to use a single unix socket to handle
+// multiple containers). This structure will probably move to runtime-spec
+// at some point. But for now it lies in libcontainer.
+type TerminalInfo struct {
+	// Version of the API.
+	Version uint32 `json:"version"`
+
+	// Type of message (future proofing).
+	Type uint8 `json:"type"`
```

Comment:
I don't see much point in separating an integer `version` from a `type` enum.  With SemVer's structured versions, the parties exchanging messages can interoperate (in some cases) even if their target versions don't match exactly.  With a non-semantic integer version that's no longer true.  Two possible approaches:

a. Make `version` a SemVer field and use it to version messages of a given type (e.g. increment it when we adjust the payload for `T`-type messages, and have a separate SemVer version for some future `X`-type (say) messages).

b. Drop `version` and make `type` a string.  A future `T`-ish message would be `T2` (or whatever).  This is the pattern used by POSIX with `dup`, `dup2`, etc.

I expect this message schema to be low-churn, so I prefer (b).

And I'd rather have less-compact types (e.g. `terminal` instead of `T`).